src/pykeen/datasets/analysis.py

```diff
@@ -301,8 +301,8 @@ def _yield_binary_patterns(
     :param pairs:
         A mapping from relations to the set of entity pairs.
 
-    :return:
-        An iterable of (relation_id, pattern_type, support, confidence) tuples.
+    :yields:
+        A tuple (relation_id, pattern_type, support, confidence).
```

Comment:
if you want really fine granular documentation for these kinds of iterables of tuples, you can always create a named tuple class and document the fields there ;)